xchange-bittrex/src/main/java/org/knowm/xchange/bittrex/service/BittrexTradeService.java

```diff
@@ -126,15 +127,19 @@ public OpenOrdersParams createOpenOrdersParams() {
         if (bittrexOrder != null) {
           Order order = BittrexAdapters.adaptOrder(bittrexOrder);
           if (order instanceof MarketOrder) {
-            List<BittrexExecution> executions = getBittrexOrderExecutions(order.getId());
-            if (executions != null) {
-              executions.stream()
-                  .findFirst()
-                  .ifPresent(
-                      execution -> {
-                        order.setAveragePrice(execution.getRate());
-                      });
-            }
+            BigDecimal rate =
+                decorateApiCall(
+                        () -> {
+                          List<BittrexExecution> executions =
+                              getBittrexOrderExecutions(order.getId());
+                          if (executions != null && executions.size() > 0) {
+                            return executions.get(0).getRate();
+                          }
+                          throw new BittrexException();
```

Comment:
I think you may probable have a do while type of statement or a for-each with you retry count, otherwise what you are trying to do is a bit obfuscated and unclear. Throwing exceptions to try and make something happen isn't really the point of exceptions, your looping logic should encompass that.

Reply:
what about now? when a market order is performed, the price is not coming from the getBittrexOrder, the retry is because if you just placed the order, it can take a few milliseconds for the execution to be created, in that case, the loop won't find anything, it will break and the retry will fetch again.